geniza/corpus/admin.py

```diff
@@ -592,3 +594,32 @@ def save_model(self, request, obj, form, change):
         # if there is an error loading the IIIF manifest
         obj.request = request
         super().save_model(request, obj, form, change)
+
+
+admin.site.unregister(Tag)
+
+
+@admin.register(Tag)
+class CustomTagAdmin(TagAdmin):
+    list_display = ("name", "slug", "document_count")
+
+    def get_queryset(self, request):
+        # The annotations we use for document count on the list view
+        # make the search too slow for autocomplete.
+        # Reset to original, unannotated queryset *only* for autocomplete
+        qs = super().get_queryset(request)
+        if request and request.path == "/admin/autocomplete/":
+            # return without annotations
+            return qs
+        # otherwise, annotate with counts
+        qs = qs.annotate(
+            # taggit_taggeditem_items is how to reference the taggeditems for tags
+            document_count=Count("taggit_taggeditem_items", distinct=True),
```

Comment:
Technically this is counting any tagged item, not just documents; we're only currently using tags for documents but I expect that to change in the future; so I suggest `item_count` for the field and `Count` for the label